lib/hanami/action.rb

```diff
@@ -410,21 +419,25 @@ def self.handle_exception(exception)
     #
     # @since 0.1.0
     # @api private
-    module InstanceMethods
-      def initialize(configuration:, **args)
-        super(**args)
-        @configuration       = configuration
-        @accepted_mime_types = Mime.restrict_mime_types(configuration, self.class.accepted_formats)
-
-        # Exceptions
-        @handled_exceptions = @configuration.handled_exceptions.merge(self.class.handled_exceptions)
-        @handled_exceptions = Hash[
-          @handled_exceptions.sort{|(ex1,_),(ex2,_)| ex1.ancestors.include?(ex2) ? -1 : 1 }
-        ]
-
-        freeze
+    def self.new(configuration:, **args)
+      allocate.tap do |obj|
+        obj.instance_variable_set(:@configuration, configuration)
+        obj.instance_variable_set(:@accepted_mime_types, Mime.restrict_mime_types(configuration, accepted_formats))
+        obj.instance_variable_set(
+          :@handled_exceptions,
+          Hash[
+            configuration
+            .handled_exceptions
+            .merge(handled_exceptions)
+            .sort{ |(ex1,_),(ex2,_)| ex1.ancestors.include?(ex2) ? -1 : 1 }
+          ]
+        )
```

Comment:
I'd love to tidy this section up in a follow-up PR, if you're open to it, @jodosha :)

Reply:
Yes, the entire gem needs a cleanup soon. But I'd punt on this for a little bit more.

From early benchmarks, this gem is slow 😢. We may want first to address these problems and then cleanup.

Reply:
No probs, happy to leave this for later.